vyper/builtins/functions.py

```diff
@@ -875,18 +876,10 @@ def build_IR(self, expr, args, kwargs, context):
         sub, index = args
         ret_type = kwargs["output_type"]
 
-        # Get length and specific element
-        if sub.location == STORAGE:
-            lengetter = IRnode.from_list(["sload", "_sub"], typ=INT128_T)
-            elementgetter = _storage_element_getter
-
-        else:
-            op = sub.location.load_op
-            lengetter = IRnode.from_list([op, "_sub"], typ=INT128_T)
-            elementgetter = _generic_element_getter(op)
+        lengetter = IRnode.from_list([sub.location.load_op, "_sub"], typ=INT128_T)
```

Comment:
let's get rid of the `typ=` here. if anything, it should be uint256, but IRnodes should only really be tagged with types if the user expression has a type.